### ISSUE TITLE
Ability to use raspistill while motioneye is running?

### ISSUE THREAD

**Issue by pghpete** [ccrisan/motioneye#41]
I am not sure this is an issue with motioneye, raspistill, or by design but I thought I would ask here.  Is it possible (or should it be possible) to run the raspberry pi raspistill camera utility when motioneye is running?  Motion/Motioneye uses /dev/video0 and I thought I was able to run the raspistill to capture still images with motioneye running in the past but I can no longer do that. [Raspberry Pi 2 Model B, Raspian Jesse, Pi NOIR camera module]

With motioneye off, "raspistill -o image.jpg" works as expected, with motioneye on, raspistill errors out with the following...

pi@seccam ~ $ raspistill -o image.jpg
mmal: mmal_vc_component_enable: failed to enable component: ENOSPC
mmal: camera component couldn't be enabled
mmal: main: Failed to create camera component
mmal: Failed to run camera app. Please check for firmware updates

I have updated Raspian and the Pi's firmware to the latest available software/firmware.

**Comment by ccrisan** [comment]
I doubt that would be possible, as the mmal interface is already in use by the v4l2 module.

**Comment by pghpete** [comment]
What a fast response lol thank you!  Perhaps a feature request to alow you to take still images from the gui is in order? :)  On a side note, is there any way I can contribute back to this project? Be it something small like a man page or documentation, or debugging, testing, etc?

**Comment by ccrisan** [comment]
Yes, the feature has already been proposed (I'm not sure if there's an actual issue about it, but I remember users asking for taking snapshots from the UI).

Well since you mentioned the key word "documentation" (:P), motionEye could really use some reference material when it comes to the settings panel. If you could take each available setting from the settings panel, per section, and document it (possibly linking to the corresponding [motion config option](http://www.lavrsen.dk/foswiki/bin/view/Motion/ConfigFileOptions), it would be great.

As soon as I have some content regarding this matter, I could turn it into a settings wiki page.

**Comment by pghpete** [comment]
Sound good. I'll spend some time and see what I can provide you.  Where should I send the resulting info and in what format?

**Comment by ccrisan** [comment]
You can send it directly to me (ccrisan around gmail) in whatever format you want. I will convert it to markdown then.

**Comment by pghpete** [comment]
fyi - sent some items via email. holler if I can improve them or update them for you.

**Comment by chrisf4lc0n** [comment]
Hi, let me chip in.
I am running a remote doorbell with my Pi 1 and planning to run MotionEye alongside. I have just thought of a workaround... I can script, that when the button on the bell gets pressed it shuts the MotionEye down, rings the bell, takes photo and restart the MotionEye.
Maybe that can help... Once I have MotionEye up and running I will report back with more details.

**Comment by chrisf4lc0n** [comment]
I had a similar problem, as I was running my Doorbell script which was taking a static photo when Doorbell was pressed, but when stream was running I could not use the camera. I sorted it out writing a simple script, for you could be a bit different but that should help:

sudo nano raspistill.sh

add:
#!/bin/sh
d=`date +%d-%m-%y`
t=`date +%T`
systemctl stop motioneye.service
raspistill -o $t$d.jpg -w 1024 -h 768 -q 30 -hf
sleep 15
systemctl start motioneye.service
exit

sudo chmod a+x raspistill.sh

That will shut the motioneye service for 15 seconds, takes photo and restarts the service.
You can obviously modify it to your needs.
If you need something more advanced let me know.

**Comment by chrisf4lc0n** [comment]
There is another way of achieving that by either using https://github.com/umlaeute/v4l2loopback or https://github.com/mpromonet/v4l2tools unfortunately the 1st one does not seem to be working with any 4.x kernel and the 2nd one does not create another device with my camera module. You can however test it and see if it works, that way you will have 2 separate camera streams with 1 camera allowing you to take photos and record/stream at the same time.

**Comment by JamBalaya56562** [comment]
Short answer: not simultaneously. On the legacy (MMAL) camera stack, the Raspberry Pi camera can only be owned by **one process at a time**. While motionEye's `motion` process has the camera open, `raspistill` will fail to acquire it (typically `mmal: Failed to create camera component` / "camera is not enabled").

A few options depending on what you actually need:

- **Just want a still image?** Use what motionEye already provides instead of `raspistill`:
  - the latest snapshot is always at `.../lastsnap.jpg` for each camera,
  - or hit the snapshot/`/picture` API endpoint,
  - or enable **Still Images** in the camera settings for periodic captures.
- **Need `raspistill` specifically?** Stop the motionEye/motion service (or disable that camera) first, run `raspistill`, then start it again — they can't share the device.
- **Need both at once?** Use a second camera (e.g. a USB webcam) for the `raspistill`-style captures, and leave the Pi camera to motionEye.

Also note that on Bookworm and later the legacy stack (and `raspistill`) is deprecated in favour of libcamera (`libcamera-still` / `rpicam-still`), and libcamera is likewise single-consumer per device — so the same one-process-at-a-time limitation applies there too.

This is a camera-stack limitation rather than a motionEye bug, and the thread has been inactive for a long time, so **this one looks safe to close**. If the built-in snapshot options don't cover your use case, let us know the details and we can take another look.

**Comment by MichaIng** [comment]
Also referring to the open issues about integrating libcamera support to motionEye via libcamerify:
- #2425
- #2683

I'm closing this old issue.
